src/macros.rs

```diff
@@ -1,199 +1,151 @@
 #[macro_export]
 macro_rules! debug {
-    ($($arg:tt)*) => (if cfg!(not(ndebug)) { log!(::log::DEBUG, $($arg)*) })
+    ($($arg:tt)*) => (
+        match () {
+            #[cfg(not(any(log_level = "off",
+                          log_level = "error",
+                          log_level = "warn",
+                          log_level = "info")))]
+            () => log!($crate::LogLevel::Debug, $($arg)*),
+            #[cfg(any(log_level = "off",
+                      log_level = "error",
+                      log_level = "warn",
+                      log_level = "info"))]
+            () => {}
+
+        }
+    )
 }
 
-/// A macro to test whether a log level is enabled for the current module.
-///
-/// # Example
-///
-/// ```
-/// #[macro_use] extern crate log;
+/// Logs a message at the trace level.
 ///
-/// struct Point { x: int, y: int }
-/// fn some_expensive_computation() -> Point { Point { x: 1, y: 2 } }
+/// Logging through this macro is disabled if any of the following cfgs are
+/// present: `log_level = "off"`, `log_level = "error"`, `log_level = "warn"`,
+/// `log_level = "info"`, or `log_level = "debug"`.
+#[macro_export]
+macro_rules! trace {
+    ($($arg:tt)*) => (
+        match () {
+            #[cfg(not(any(log_level = "off",
+                          log_level = "error",
+                          log_level = "warn",
+                          log_level = "info",
+                          log_level = "debug")))]
+            () => log!($crate::LogLevel::Trace, $($arg)*),
+            #[cfg(any(log_level = "off",
+                      log_level = "error",
+                      log_level = "warn",
+                      log_level = "info",
+                      log_level = "debug"))]
+            () => {}
+        }
+    )
+}
+
+/// Determines if a message logged at the specified level in that module will
+/// be logged.
 ///
-/// fn main() {
-///     if log_enabled!(log::DEBUG) {
-///         let x = some_expensive_computation();
-///         debug!("x.x = {}, x.y = {}", x.x, x.y);
-///     }
-/// }
-/// ```
+/// This can be used to avoid expensive computation of log message arguments if
+/// the message would be ignored anyway.
 ///
-/// Assumes the binary is `main`:
+/// # Examples
 ///
-/// ```{.bash}
-/// $ RUST_LOG=error ./main
-/// ```
+/// ```rust
+/// # #[macro_use]
+/// # extern crate log;
+/// use log::LogLevel::Debug;
 ///
-/// ```{.bash}
-/// $ RUST_LOG=debug ./main
-/// DEBUG:main: x.x = 1, x.y = 2
+/// # fn foo() {
+/// if log_enabled!(Debug) {
+///     debug!("expensive debug data: {}", expensive_call());
+/// }
+/// # }
+/// # fn expensive_call() -> u32 { 0 }
+/// # fn main() {}
 /// ```
 #[macro_export]
 macro_rules! log_enabled {
     ($lvl:expr) => ({
         let lvl = $lvl;
-        (lvl != ::log::DEBUG || cfg!(not(ndebug))) &&
-        lvl <= ::log::log_level() &&
-        ::log::mod_enabled(lvl, module_path!())
+        lvl <= $crate::max_log_level() && $crate::enabled(lvl, module_path!())
```

Comment:
Issue #3 goes into some detail about this:
> The logging macros don't call enabled before calling log. It turned out when I was experimenting with logger implementations that log can pretty trivially perform the filtering itself. If argument computation is expensive, the log_enabled! macro is still available, which will call through to enabled.

For log4rs in particular, `enabled` basically does the same thing as `log`, just returning a bool instead of performing the actual write. So unless the computation of the log arguments is expensive, just calling `log` instead of `enabled` and then maybe `log` should end up being a perf win.